custom_components/adaptive_lighting/switch.py

```diff
@@ -567,14 +632,38 @@ def __init__(
         self._name = data[CONF_NAME]
         self._lights = data[CONF_LIGHTS]
 
+        for light in data[CONF_LIGHTS]:
+            if not hasattr(self.turn_on_off_listener, "_switches"):
+                self.turn_on_off_listener._switches = {}
+            self.turn_on_off_listener._switches[
+                light
+            ] = self  # light should never be in multiple switches anyway.
+
+        self._take_over_control = data[CONF_TAKE_OVER_CONTROL]
+        self._alt_detect_method = data[CONF_ALT_DETECT_METHOD]
+        self._strict_adapting = data[CONF_STRICT_ADAPTING]
         self._detect_non_ha_changes = data[CONF_DETECT_NON_HA_CHANGES]
+        if not data[CONF_TAKE_OVER_CONTROL] and (
+            not data[CONF_STRICT_ADAPTING]
+            or data[CONF_ALT_DETECT]
```

Comment:
This variable `CONF_ALT_DETECT` is not defined. If you never ran into this issue it means this code never got executed.